vignettes/admiral.Rmd

```diff
@@ -19,74 +19,101 @@ library(admiraldev)
+offered by `{admiral}`, as well as the argument conventions they follow and how best 
+to start an `{admiral}` script. 
 
-First, we will load our packages and example datasets to help with our `ADSL` creation. The `{dplyr}` and `{lubridate}` packages are `{tidyverse}` packages and used heavily throughout this script.  The `{admiral}` package also leverages the `{pharmaversesdtm}` package for example SDTM datasets which are from the CDISC Pilot Study.
+# Setup
+
+To help with the examples showcased, we will load some packages and set up some example 
+data to manipulate later. The `{dplyr}`, `{lubridate}` and `{stringr}` packages are 
+`{tidyverse}` packages and are used heavily throughout this vignette. The `{admiral}`
+package also leverages the `{pharmaversesdtm}` package for example SDTM datasets 
+which are from the CDISC Pilot Study.
 
 ```{r, message=FALSE, warning=FALSE}
+# Uncomment line below if you need to install these packages
+# install.packages(c("dplyr", "lubridate", "stringr", "tibble", "pharmaversesdtm", "admiral"))
+
 library(dplyr, warn.conflicts = FALSE)
 library(lubridate)
-library(admiral)
+library(stringr)
+library(tibble)
 library(pharmaversesdtm)
+library(admiral)
 
 # Read in SDTM datasets
 data("dm")
 data("ds")
 data("ex")
-
-dm <- convert_blanks_to_na(dm)
-ds <- convert_blanks_to_na(ds)
-ex <- convert_blanks_to_na(ex)
-```
-
-## Derive Treatment Variables (`TRT0xP`, `TRT0xA`) {#treatmentvargs}
-
-The mapping of the treatment variables is left to the ADaM programmer. An example mapping may be:
-
-```{r eval=TRUE}
-adsl <- dm %>%
-  mutate(TRT01P = ARM, TRT01A = ACTARM)
+data("vs")
+data("admiral_adsl")
 ```
 
-## Derive/Impute Numeric Treatment Date/Time and Flag Variables (`TRTSDTM`, `TRTEDTM`, `TRTSTMF`, `TRTETMF`) {#trtdatetimegs}
-
-The function `derive_vars_dtm()` can be used to convert the DTC variables from
-`EX` to numeric datetime variable and impute missing components. The function
-call returns the original data frame with the column `EXSTDTM`, `EXENDTM` and
-corresponding time imputation  flag variables `EXSTTMF` and  `EXENTMF` added to
-the end of the dataframe. Exposure observations with an incomplete date are
-ignored. We impute missing time to be 23:59:59 using `time_imputation = "last"`.
-The required imputation flags are determined automatically by the function. Here
-only time imputation flags are derived because time is imputed but date is not
-imputed.
-
-Don't be intimidated by the number of arguments! We try to make our arguments
-self-explanatory, e.g. the `new_vars_prefix` places `EXST` at the start of the
-`--DTM` variable and `time_imputation = "last"` appends 23:59:59.  However, this
-is not always possible to make every argument self-explanatory.  If you click on
-the function, `derive_vars_dtm()`, you can bring up the reference documentation
-and learn more about each argument.
+The `adsl` and `advs` objects will now be prepared to showcase [addition of variables](#addvars)
```

Comment:
Can we add a note to refer to the Handling missing data section ? I found it useful to explain on the "Get started" vignette why we have to convert the blanks to NA: the {admiral} functions don't do it automatically. I think it's one of the basics to know, especially if users are also new to R (not only to {admiral}) it was a good reminder for a good start.

Reply:
@Fanny-Gautier added a note

Reply:
Nice, thanks